doc/src/manual/types.md

```diff
@@ -497,7 +497,18 @@ ERROR: TypeError: in typeassert, expected Union{Int64, AbstractString}, got Floa
 ```
 
 The compilers for many languages have an internal union construct for reasoning about types; Julia
-simply exposes it to the programmer.
+simply exposes it to the programmer. The Julia compiler is able to generate efficient code in the
+presence of `Union` types with a small number of types[^smallunion], by generating specialized code
+in separate branches for each possible type.
+
+[^smallunion]: "Small" is defined by the `MAX_UNION_SPLITTING` constant, which is currently set to 4.
```

Comment:
Turns out that's not the valid syntax for footnotes. Looking at parallel-computing.md, it seems the note has to be at the end, with a four-space indentation. I'll fix it with other changes to avoid running CI just for that.

Reply:
Fun fact: you can put `[ci skip]` in your commit message to avoid triggering CI. Travis, AppVeyor, and CircleCI all support it natively, and even the Julia FreeBSD CI has support: https://github.com/JuliaLang/julia/issues/20585#issuecomment-315508232

Reply:
Yeah, but I also hope somebody can confirm what's the right syntax for footnotes. ;-)

Reply:
I've pushed a commit to fix the footnote syntax, reflecting my understanding of how it works.